src/Extensions/Diagnostics.HealthChecks/ServiceCollectionExtensions.cs

```diff
@@ -45,5 +62,30 @@ public static IServiceCollection AddHealthCheckPublisher<TPublisher>(this IServi
 			serviceCollection.TryAddEnumerable(ServiceDescriptor.Singleton<IHealthCheckPublisher, TPublisher>());
 			return serviceCollection;
 		}
+
+		/// <summary>
+		/// 
+		/// </summary>
+		/// <param name="serviceCollection"></param>
+		/// <returns></returns>
+		public static IServiceCollection AddRestHealthCheckPublisher(this IServiceCollection serviceCollection)
+		{
+			serviceCollection.TryAddEnumerable(ServiceDescriptor.Singleton<IHealthCheckPublisher, RestHealthCheckPublisher>(
+				(serviceProvider) =>
+				{
+					return new RestHealthCheckPublisher(
+						clusterEndpoints: new Uri(serviceProvider
+													.GetRequiredService<IOptions<RestHealthCheckPublisherOptions>>()
+													.Value
+													.RestHealthPublisherClusterEndpoint),
+						serviceId: serviceProvider
+													.GetRequiredService<IOptions<RestHealthCheckPublisherOptions>>()
+													.Value
+													.RestHealthPublisherServiceId
```

Comment:
```suggestion
					RestHealthCheckPublisherOptions options = serviceProvider.GetRequiredService<IOptions<RestHealthCheckPublisherOptions>>().Value;
					return new RestHealthCheckPublisher(
						clusterEndpoints: new Uri(options.RestHealthPublisherClusterEndpoint),
						serviceId: options.RestHealthPublisherServiceId
```

Reply:
Also it might be better just get `IOptions<RestHealthCheckPublisherOptions>` as constructer parameter for `RestHealthCheckPublisher`, then you can just register type.